### PR TITLE
Tweaking PanelsChildGestureRegionObserver

### PR DESCRIPTION
**Pull request by charlesx2013**
This PR fixes an issue where the `PanelsChildGestureRegionObserver` wasn't working at all for integrated viewpagers due to the bounds not being updated correctly when it was scrolling up or down.

https://user-images.githubusercontent.com/5563367/129962435-41ae96e1-ddb8-4a1a-af16-d4fe890ee037.mp4

@donaldchen

### CODE REVIEW

File: overlapping_panels/src/main/java/com/discord/panels/PanelsChildGestureRegionObserver.kt [hunk 1]
```diff
@@ -42,39 +45,74 @@ class PanelsChildGestureRegionObserver : View.OnLayoutChangeListener {
...
+    )
+
+    publishGestureRegionsUpdate()
+  }
+
+  @UiThread
+  fun register(view: View) {
+    view.addOnLayoutChangeListener(this)
+
+    val listener = ViewTreeObserver.OnScrollChangedListener {
```
Comment (donaldchen) [inline overlapping_panels/src/main/java/com/discord/panels/PanelsChildGestureRegionObserver.kt:71]: To make sure I understand this, does this mean that previously, we weren't updating the panel child gesture region on scrolls?

File: overlapping_panels/src/main/java/com/discord/panels/PanelsChildGestureRegionObserver.kt [hunk 2]
```diff
@@ -42,39 +45,74 @@ class PanelsChildGestureRegionObserver : View.OnLayoutChangeListener {
...
   fun remove(viewId: Int) {
     viewIdToGestureRegionMap.remove(viewId)
     publishGestureRegionsUpdate()
   }
 
+  /**
+   * Stop publishing gesture region updates based on layout and scroll changes to android.view.View
+   */
+  @UiThread
+  fun remove(view: View) {
```
Comment (donaldchen) [inline overlapping_panels/src/main/java/com/discord/panels/PanelsChildGestureRegionObserver.kt:107]: Should this be named `unregister` based on the deprecation message on the other `remove` method?

**Review by donaldchen — COMMENTED** [review]
The high level changes make sense! Are you down to add a view pager inside the sample app, so we can verify that this works within this repo?

File: overlapping_panels/src/main/java/com/discord/panels/PanelsChildGestureRegionObserver.kt (hunk 2 above)
Comment (charlesx2013) [reply overlapping_panels/src/main/java/com/discord/panels/PanelsChildGestureRegionObserver.kt:107]: whoops, good catch

File: overlapping_panels/src/main/java/com/discord/panels/PanelsChildGestureRegionObserver.kt [hunk 3]
```diff
@@ -42,39 +45,74 @@ class PanelsChildGestureRegionObserver : View.OnLayoutChangeListener {
...
+    viewIdToListenerMap.remove(view.id)?.let {
+      view.viewTreeObserver.removeOnScrollChangedListener(it)
+    }
```
Comment (donaldchen) [inline overlapping_panels/src/main/java/com/discord/panels/PanelsChildGestureRegionObserver.kt:110]: Previously, `addOnLayoutChangeListener` was called outside of this class. Now that we're calling it inside this class, should we also call `removeOnLayoutChangeListener` in this `remove` / `unregister` method?

File: overlapping_panels/src/main/java/com/discord/panels/PanelsChildGestureRegionObserver.kt (hunk 1 above)
Comment (charlesx2013) [reply overlapping_panels/src/main/java/com/discord/panels/PanelsChildGestureRegionObserver.kt:71]: yep! just on layout change.

File: sample_app/build.gradle [hunk 4]
```diff
@@ -57,6 +61,8 @@ dependencies {
     implementation "androidx.core:core-ktx:$androidxCoreVersion"
     implementation "io.reactivex.rxjava3:rxjava:$rxJavaVersion"
     implementation project(':overlapping_panels')
+    implementation 'androidx.coordinatorlayout:coordinatorlayout:1.1.0'
```
Comment (donaldchen) [inline sample_app/build.gradle:64]: nit: Can we create a variable for `1.1.0` and `1.4.0` similar to what we do with `rxJavaVersion` and `androidxCoreVersion`?

**Review by donaldchen — APPROVED** [review]
Looks great!

**Pull request merged**